File: solutions/2639-separate-the-digits-in-an-array/solution.py
class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
      a=[]
      for i  in range(len(nums)-1,-1,-1):
        while nums[i]!=0:
            a.append(nums[i]%10)
            nums[i]//=10
      return a[::-1]
            

File: solutions/2639-separate-the-digits-in-an-array/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* separateDigits(int* nums, int size, int* rise) {
    int i; 
    int dig = 0; 
    for(i = 0; i < size; i++){
        int temp = nums[i]; 
        while(temp != 0){
            dig++; 
            temp /= 10;
        }
    }
    int index = dig-1;
    int *arr = (int*)malloc(dig*sizeof(int)); 
    for(i = size -1; i >= 0; i--){
        int temp = nums[i]; 
        while(temp != 0){
            arr[index] = temp%10; 
            index--;
            temp /=10;
        }
    }
    *rise = dig; 
    return arr; 
}
